qutip/visualization.py

```diff
@@ -935,34 +974,33 @@ def plot_energy_levels(H_list, N=0, labels=None, show_ylabels=False,
 
         evals0 = evals1
 
-    ax.set_frame_on(False)
-
-    if show_ylabels:
+    if energy_levels:
         yticks = np.unique(np.around(yticks, 1))
-        ax.set_yticks(yticks)
+        _set_ticklabels(ax, energy_levels, yticks, 'y')
     else:
-        ax.axes.get_yaxis().set_visible(False)
+        # show eigenenergies
+        yticks = np.unique(np.around(yticks, 1))
+        ax.set_yticks(yticks)
 
-    if labels:
+    if h_labels:
         ax.get_xaxis().tick_bottom()
-        ax.set_xticks(xticks)
-        ax.set_xticklabels(labels, fontsize=16)
+        _set_ticklabels(ax, h_labels, xticks, 'x')
     else:
-        ax.axes.get_xaxis().set_visible(False)
+        # hide xtick
+        ax.tick_params(axis='x', which='both',
+                       bottom=False, labelbottom=False)
 
     return fig, ax
 
 
 def energy_level_diagram(H_list, N=0, labels=None, show_ylabels=False,
                          figsize=(8, 12), fig=None, ax=None):
     warnings.warn("Deprecated: Use plot_energy_levels")
-    return plot_energy_levels(H_list, N=N, labels=labels,
-                              show_ylabels=show_ylabels,
-                              figsize=figsize, fig=fig, ax=ax)
+    return plot_energy_levels(H_list, N=N, fig=fig, ax=ax)
 
 
-def plot_fock_distribution(rho, offset=0, fig=None, ax=None,
-                           figsize=(8, 6), title=None, unit_y_range=True):
+def plot_fock_distribution(rho, offset=0, unit_y_range=True, *,
```

Comment:
`offset` just control the x label, maybe change it for that.
The color, alpha could be options inputs.

Reply:
I think ```offset``` is good. Shifting x ticklabels with it is easier than shifting x ticklabels by specifying their positions.
I added ```color``` option. Thanks!
I will not add alpha. I do not know when users want to change alpha.

Reply:
But this only allows to shift labels, not use more useful ones. We never actually check that the state is a single fock state and it could be used with tensored space if we allow customs labels.

Reply:
Now, users can set xticklabels.
<img width="625" alt="fock" src="https://github.com/qutip/qutip/assets/72233550/712aa5b2-002f-4ade-af26-9cfa247f4f24">